fetch/client/src/main/scala/endpoints4s/fetch/ChunkedEntities.scala

```diff
@@ -0,0 +1,118 @@
+package endpoints4s.fetch
+
+import endpoints4s.algebra
+import org.scalajs.dom
+import org.scalajs.dom.ReadableStreamReader
+
+import scala.scalajs.js
+import scala.scalajs.js.typedarray.Uint8Array
+import scala.scalajs.js.|
+
+trait ChunkedResponseEntities
+    extends algebra.ChunkedResponseEntities
+    with EndpointsWithCustomErrors {
+
+  type Chunks[A] = dom.ReadableStream[A]
+
+  def textChunksResponse: ResponseEntity[Chunks[String]] =
+    chunkedResponseEntity(uint8array => Right(new TextDecoder("utf-8").decode(uint8array)))
+
+  def bytesChunksResponse: ResponseEntity[Chunks[Array[Byte]]] =
+    chunkedResponseEntity(uint8array => Right(uint8array.toArray.map(_.toByte)))
+
+  private[fetch] def chunkedResponseEntity[A](
+      fromUint8Array: Uint8Array => Either[Throwable, A],
+      framing: dom.ReadableStream[Uint8Array] => dom.ReadableStream[Uint8Array] = identity
+  ): ResponseEntity[Chunks[A]] = { response =>
+    val readableStream = ReadableStream[A](
+      new ReadableStreamUnderlyingSource[A] {
+        start = js.defined((controller: ReadableStreamController[A]) => {
+          def read(reader: ReadableStreamReader[Uint8Array]): js.Promise[Unit] = {
+            reader
+              .read()
+              .`then`((chunk: dom.Chunk[Uint8Array]) => {
+                if (chunk.done) {
+                  controller.close(): Unit | js.Thenable[Unit]
+                } else {
+                  fromUint8Array(chunk.value)
+                    .fold(
+                      error => js.Promise.reject(error),
+                      value => {
+                        controller.enqueue(value)
+                        read(reader): Unit | js.Thenable[Unit]
+                      }
+                    ): Unit | js.Thenable[Unit]
+                }
+              })
+          }
+          read(framing(response.body).getReader()): js.UndefOr[js.Promise[Unit]]
+        }): js.UndefOr[js.Function1[ReadableStreamController[A], js.UndefOr[js.Promise[Unit]]]]
+      }
+    )
+    js.Promise.resolve[Either[Throwable, dom.ReadableStream[A]]](
+      Right(readableStream.asInstanceOf[dom.ReadableStream[A]])
+    )
+  }
+}
+
+trait ChunkedJsonResponseEntities
+    extends algebra.ChunkedJsonResponseEntities
+    with ChunkedResponseEntities
+    with JsonEntitiesFromCodecs {
+
+  type Framing = dom.ReadableStream[Uint8Array] => dom.ReadableStream[Uint8Array]
```

Comment:
Now this only represents response chunk framing, because requests are not yet supported by browsers, but maybe it would make sense to make the type more flexible here, so we would not be required to introduce a breaking change once the browser support finally comes around? Or does it not make a lot of sense to think this far ahead? 😄

Reply:
It is OK to think far ahead :)

So far, we used to be less conservative about breaking changes in interpreters, though, but it’s always good (for the end-users) to reduce the amount of breaking changes!

You could do this to support future extensions:

```suggestion
  class Framing(val response: dom.ReadableStream[Uint8Array] => dom.ReadableStream[Uint8Array])
```

Reply:
And this type does not need to be moved to its own trait?

Reply:
I don’t think it _needs_ to, but feel free to do so, if you think that will help!